src/wp-includes/class-wp-term-query.php

```diff
@@ -836,54 +841,24 @@ public function get_terms() {
 			$terms = $_terms;
 		}
 
-		$_terms = array();
-		if ( 'id=>parent' === $_fields ) {
-			foreach ( $terms as $term ) {
-				$_terms[ $term->term_id ] = $term->parent;
-			}
-		} elseif ( 'ids' === $_fields ) {
-			foreach ( $terms as $term ) {
-				$_terms[] = (int) $term->term_id;
-			}
-		} elseif ( 'tt_ids' === $_fields ) {
-			foreach ( $terms as $term ) {
-				$_terms[] = (int) $term->term_taxonomy_id;
-			}
-		} elseif ( 'names' === $_fields ) {
-			foreach ( $terms as $term ) {
-				$_terms[] = $term->name;
-			}
-		} elseif ( 'slugs' === $_fields ) {
-			foreach ( $terms as $term ) {
-				$_terms[] = $term->slug;
-			}
-		} elseif ( 'id=>name' === $_fields ) {
-			foreach ( $terms as $term ) {
-				$_terms[ $term->term_id ] = $term->name;
-			}
-		} elseif ( 'id=>slug' === $_fields ) {
-			foreach ( $terms as $term ) {
-				$_terms[ $term->term_id ] = $term->slug;
-			}
-		}
-
-		if ( ! empty( $_terms ) ) {
-			$terms = $_terms;
-		}
-
```

Comment:
I don't see this logic check in the new `format_terms()` method. Is that intentional or an oversight?

Reply:
> Why is this line needed?

I'm not sure, I was hoping you knew? ;)

Reply:
The check was introduced in https://core.trac.wordpress.org/changeset/25161. It's not obvious to me that it's necessary. It looks to me like `$_terms` can only be an empty array if `$terms` (`$terms_objects` in the new `format_terms()` method) was also an empty array, or if there's no matching `elseif` clause, which would also result in an empty array.

Reply:
I think we should leave this out for now. None of the tests flagged this as an issue.